### PR TITLE
Closes #1595 (edit discussion)

### LINKED ISSUE #1595: Discussion - Edit

### PR DESCRIPTION
**Pull request by pepoospina**

### CODE REVIEW

File: functions/src/discussion/controllers/discussion/update.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        true
+      );
+
+      if (discussion.ownerId !== req.user.uid)
+        throw new CommonError("Not discussion owner", {
+          ownerId: req.user.uid,
+          discussionId: validated.discussionId,
+        });
+
+      // circle visibility should remain the same
```
Comment (ariel-dao) [inline functions/src/discussion/controllers/discussion/update.ts]: This should be inferred from the type (validation schema), remove comment

File: functions/src/discussion/db/discussion/updateDiscussion.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { discussionCollection } from './discussionCollection';
 import { Discussion } from '../../../shared/interfaces/discussions/Discussion';
 import { NotFoundError } from '../../../util/errors';
 
-type UpdateDiscussionPayload = Partial<Pick<Discussion, 'lastMessage' | 'circleVisibility'>>
+export type UpdateDiscussionPayload = Partial<Pick<Discussion, 'lastMessage' | 'circleVisibility' | 'title' | 'message'>>
```
Comment (ariel-dao) [inline functions/src/discussion/db/discussion/updateDiscussion.ts:6]: Circle visibility update endpoint and update endpoint should accept different types, you can user InferType from yup to use the validaiton schema as the type

File: functions/src/shared/schemas/updateDiscussion.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+export const updateDiscussionDataValidationScheme = object({
+  discussionId: string().required(),
+
+  title: string().required(),
+
+  message: string().required(),
+});
+
+export type UpdateDiscussion = InferType<
+  typeof updateDiscussionDataValidationScheme
```
Comment (ariel-dao) [inline functions/src/shared/schemas/updateDiscussion.ts:13]: This should be used as the type for update discussion

File: functions/src/discussion/controllers/discussion/update.ts (hunk 1 above)
Comment (pepoospina) [reply functions/src/discussion/controllers/discussion/update.ts]: Why do you want me to remove the comment? This is a reminder that the circle-visibility update flow is not included in this discussion update service and thus is forced to remain the same. I will update the comment

File: functions/src/discussion/db/discussion/updateDiscussion.ts (hunk 2 above)
Comment (pepoospina) [reply functions/src/discussion/db/discussion/updateDiscussion.ts:6]: You mean that we should have two db functions to update discussions too? One to update properties and the other one to update visibility? I kept one on purpose. I prefer to encapsulate and reuse DB operations as much as possible.

File: functions/src/shared/schemas/updateDiscussion.ts (hunk 3 above)
Comment (pepoospina) [reply functions/src/shared/schemas/updateDiscussion.ts:13]: Not sure what you mean.

**Comment by pepoospina** [comment]
A single PR to edit and hide discussion will be this one: https://github.com/daostack/common-backend/pull/1637

**Review by pepoospina — COMMENTED** [review]
I had not submitted my comments!!! 
Sorry, didn't know I hade to submit the review

**Pull request closed without merging**